src/comment_worker.py

```diff
@@ -603,33 +688,56 @@ def fire(self, sess, comment, investor, to_fire):
         if investor.firm == 0:
             return comment.reply_wrap(message.firm_none_org)
 
-        if investor.firm_role == "":
-            return comment.reply_wrap(message.not_ceo_or_exec_org)
-
         user = sess.query(Investor).\
             filter(func.lower(Investor.name) == func.lower(to_fire)).\
             first()
         if (user == None) or (user.name == investor.name) or (user.firm != investor.firm):
             return comment.reply_wrap(message.fire_failure_org)
 
-        if (investor.firm_role != "ceo") and (user.firm_role != ""):
-            return comment.reply_wrap(message.not_ceo_org)
+        #if (investor.firm_role != "ceo") and (user.firm_role != ""):
+        #    return comment.reply_wrap(message.not_ceo_org)
 
         firm = sess.query(Firm).\
             filter(Firm.id == investor.firm).\
             first()
 
-        firm.size -= 1
-        if user.firm_role == 'exec':
+        if user.firm_role == "":
+            if (investor.firm_role == "") or (investor.firm_role == "assoc"):
+                return comment.reply_wrap(message.not_ceo_or_exec_org)
+
+        elif user.firm_role == "assoc":
+            if (investor.firm_role == "") or (investor.firm_role == "assoc"):
+                return comment.reply_wrap(message.not_ceo_or_coo_org)
+
+            firm.assocs -= 1
+
+        elif user.firm_role == "exec":
+            if (investor.firm_role != "ceo") or (investor.firm_role != "coo"):
+                return comment.reply_wrap(message.not_ceo_or_coo_org)
+
             firm.execs -= 1
 
+        elif user.firm_role == "cfo":
+            if (investor.firm_role != "ceo"):
+                return comment.reply_wrap(message.not_ceo_org)
+
+            firm.cfo -= 1
+
+        elif user.firm_role == "coo":
+            if (investor.firm_role != "ceo"):
+                return comment.reply_wrap(message.not_ceo_org)
+
+            firm.coo -= 1
+
+        firm.size -= 1
```

Comment:
Lines 690 through 732: Add new roles to "!fire" command as well as adjusting role permissions